src/test/java/org/apache/sling/scripting/sightly/impl/engine/extension/FormatFilterExtensionTest.java

```diff
@@ -0,0 +1,178 @@
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.script.Bindings;
+import javax.script.SimpleBindings;
+
+import org.apache.sling.scripting.sightly.SightlyException;
+import org.apache.sling.scripting.sightly.render.AbstractRuntimeObjectModel;
+import org.apache.sling.scripting.sightly.render.RenderContext;
+import org.apache.sling.scripting.sightly.render.RuntimeObjectModel;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+public class FormatFilterExtensionTest {
+
+    private final RenderContext renderContext = new RenderContext() {
+        @Override public RuntimeObjectModel getObjectModel() {
+            return new AbstractRuntimeObjectModel() {
+            };
+        }
+
+        @Override public Bindings getBindings() {
+            return new SimpleBindings();
+        }
+
+        @Override public Object call(String s, Object... objects) {
+            return null;
+        }
+    };
+    private final FormatFilterExtension subject = new FormatFilterExtension();
+    private final Date testDate = Date.from(LocalDateTime.of(1918, 12, 1, 0, 0, 0, 0)
+        .atZone(ZoneId.of("UTC"))
+        .toInstant());
+
+    @Test
+    public void testDateFormatNull() {
+        assertNull(subject.call(renderContext, "default", new HashMap<String, Object>() {{
+            put(FormatFilterExtension.TYPE_OPTION, "date");
+            put(FormatFilterExtension.FORMAT, null);
+        }}));
+    }
+
+    @Test
+    public void testDateFormatNoDateObject() {
+        assertNull(subject.call(renderContext, "yyyy-MM-dd", Collections.singletonMap(FormatFilterExtension.FORMAT, new Object())));
+    }
+
+    @Test(expected = SightlyException.class)
+    public void testDateFormatFalseFormat() {
+        assertDate(null, "yT", null, null);
+    }
+
+    @Test
+    public void testDateFormatWithUTC() {
+        assertDate("1918-12-01 00:00:00.000Z", "yyyy-MM-dd HH:mm:ss.SSSXXX", "UTC", null);
+    }
+
+    @Test
+    public void testDateFormatWithZoneOffset() {
+        assertDate("1918-12-01 02:00:00.000+02:00", "yyyy-MM-dd HH:mm:ss.SSSXXX", "GMT+02:00", null);
+    }
+
+    @Test
+    public void testDateFormatWithZoneOffsetRFC822() {
+        assertDate("1918-12-01 02:00:00.000+0200", "yyyy-MM-dd HH:mm:ss.SSSZ", "GMT+02:00", null);
+    }
+
+    @Test
+    public void testDateFormatWithZoneName() {
+        assertDate("1918-12-01 02:00:00.000(GMT+02:00)", "yyyy-MM-dd HH:mm:ss.SSS(z)", "GMT+02:00", null);
+    }
+
+    /**
+     * When using jdk9 or newer, make sure to set the {@code java.locale.providers = COMPAT,SPI}
+     *
+     * @see <a href="https://docs.oracle.com/javase/9/docs/api/java/util/spi/LocaleServiceProvider.html">LocaleServiceProvider</a>
+     */
+    @Test
+    public void testDateFormatWithEscapedCharacters() {
+        assertDate("01 December '18 12:00 AM; day in year: 335; week in year: 49",
```

Comment:
Can we work with https://junit.org/junit4/javadoc/4.12/org/junit/Assume.html here to make sure that COMPAT is being active? Probably this requires a check for either `System.getProperty("java.version")` smaller than 9 or `System.getProperty("java.locale.providers") starting with "COMPAT". WDYT?

Reply:
Makes sense, yes.